modules/installation-special-config-kmod.adoc

```diff
@@ -0,0 +1,326 @@
+To add the MachineConfig later via the MCO instead, the ssh public key is not required.
+For both type, the example simple-kmod service creates a systemd unit file,
+which requires a `kmods-via-containers@simple-kmod.service`.
+
+[NOTE]
+====
+The systemd unit is a workaround for an
+link:https://github.com/coreos/ignition/issues/586[upstream bug]
+and makes sure that the `kmods-via-containers@simple-kmod.service` gets started
+on boot:
+====
+
+. Get a RHEL 8 system, then register and subscribe it:
++
+----
+# subscription-manager register
+Username: yourname
+Password: ***************
+# subscription-manager subscribe --auto
+----
+
+. Install software needed to build the software:
++
+----
+# yum install podman make git -y
+----
+
+. Create an Ignition config file that creates a systemd unit file:
++
+----
+$ mkdir kmods; cd kmods
+$ cat <<EOF > ./baseconfig.ign
+{
+  "ignition": { "version": "2.2.0" },
+  "passwd": {
+    "users": [
+      {
+        "name": "core",
+        "groups": ["sudo"],
+        "sshAuthorizedKeys": [
+          "ssh-rsa AAAA"
+        ]
+      }
+    ]
+  },
+  "systemd": {
+    "units": [{
+      "name": "require-kvc-simple-kmod.service",
+      "enabled": true,
+      "contents": "[Unit]\nRequires=kmods-via-containers@simple-kmod.service\n[Service]\nType=oneshot\nExecStart=/usr/bin/true\n\n[Install]\nWantedBy=multi-user.target"
+    }]
+  }
+}
+EOF
+----
++
+[NOTE]
+====
+You must add your public SSH key to the `baseconfig.ign` file
+to use the file during `openshift-install`.
+The public SSH key is not needed if you create the MachineConfig via the MCO.
+====
+
+. Create a base MCO YAML snippet that looks like:
+
+----
+$ cat <<EOF > mc-base.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 10-kvc-simple-kmod
+spec:
+  config:
+EOF
+----
+
+. Get the `kmods-via-containers` software:
++
+----
+$ git clone https://github.com/kmods-via-containers/kmods-via-containers
+$ git clone https://github.com/kmods-via-containers/kvc-simple-kmod
+----
+
+. Get your module software. In this example, `kvc-simple-kmod` is used:
+
+. Create a fakeroot directory and populate it with files that you want to
+deliver via Ignition, using the repositories cloned earlier:
++
+----
+$ FAKEROOT=$(mktemp -d)
+$ cd kmods-via-containers
+$ make install DESTDIR=${FAKEROOT}/usr/local CONFDIR=${FAKEROOT}/etc/
+$ cd ../kvc-simple-kmod
+$ make install DESTDIR=${FAKEROOT}/usr/local CONFDIR=${FAKEROOT}/etc/
+----
+
+. Use a tool call the `filetranspiler` to generate a final MachineConfig YAML (`mc.yaml`),
+and have it include the base Ignition config, base MachineConfig, and the fakeroot directory
+with files you would like to deliver:
++
+----
+$ ./filetranspiler/filetranspile -i ./baseconfig.ign \
+     -f ${FAKEROOT} --format=yaml --dereference-symlinks \
+     | sed 's/^/     /' | (cat mc-base.yaml -) > 99_simple-kmod.yaml
+----
+
```

Comment:
`99_simple-kmod.yaml` is only applied to worker nodes.  Can we add directions or a note that they must create a kmod for the master node?  The difference being `machineconfiguration.openshift.io/role: master` instead of `machineconfiguration.openshift.io/role: worker`

Reply:
I added this note:
The `mc-base.yaml` is set to deploy the kernel module on `worker` nodes.
To deploy on master nodes, change the role from `worker` to `master`.
To do both, you could repeat the whole procedure using different file names
for the two types of deployments.